### PR TITLE
coap fota releated fixes

### PR DESCRIPTION
**Pull request by chmielewskiandreas**

### CODE REVIEW

File: subsys/net/lib/download_client/src/download_client.c [hunk 1]
```diff
@@ -433,7 +433,9 @@ static size_t socket_recv(struct download_client *dl)
 	case IPPROTO_DTLS_1_2:
 		if (IS_ENABLED(CONFIG_COAP)) {
 			timeout = coap_get_recv_timeout(dl);
-			if (timeout <= 0) {
+			if (timeout < 0) {
+				return -1;
```
Comment (rlubos) [inline subsys/net/lib/download_client/src/download_client.c:437]: Should this set a custom errno then? Otherwise the behaviour is unpredictable.

Comment (chmielewskiandreas) [reply subsys/net/lib/download_client/src/download_client.c]: thx, i think an assert is the better choice here

**Review by rlubos — APPROVED** [review]

**Comment by andreas-chmielewski-gcx** [comment]
I see that this PR is not yet merged. It is possible to merge this in the next release?

**Comment by rlubos** [comment]
@lemrey Please have a look.

**Comment by andreas-chmielewski-gcx** [comment]
> @lemrey Please have a look.

Can you please check this patch so that FOTA using CoAP works again :) thx

**Comment by chmielewskiandreas** [comment]
Can somebody please take a look at this patch? FOTA is still broken for 2.0 ...

**Comment by lemrey** [comment]
@chmielewskiandreas I have rebased your PR and changed
```
-#include <sys/__assert.h>
+#include <zephyr/sys/__assert.h>
```
in the first commit. Added a milestone to get it into NCS v2.1.0

**Review by lemrey — APPROVED** [review]

**Comment by lemrey** [comment]
@chmielewskiandreas, could you please write a couple of lines for the release changelog? :)

**Comment by lemrey** [comment]
I have updated the release notes for this patch. @divipillai could you please have a look?

**Review by divipillai — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -687,6 +687,11 @@ Libraries for networking
     * The library has been reworked to use `Azure SDK for Embedded C`_.
     * The APIs for both IoT Hub and DPS interaction have changed and the applications and samples that use the library have been updated.
 
+   * :ref:`lib_download_client` library:
+
+     * Fixed handling of duplicated CoAP packets.
+     * Fixed handling of timeout errors when using CoAP.
+
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:694]:
```suggestion
  * :ref:`lib_download_client` library:

    * Fixed handling of duplicated CoAP packets.
    * Fixed handling of timeout errors when using CoAP.

```

**Review by divipillai — APPROVED** [review]
Please fix the alignment issue.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (lemrey) [reply doc/nrf/releases/release-notes-changelog.rst]: Thank you, updated

**Comment by lemrey** [comment]
Tried a rebase

**Comment by rlubos** [comment]
`fw-update` plan seems to be unstable, it passed in one of the preivious run - merging.

**Pull request merged**